internal/local/structure_v1.go

```diff
@@ -110,3 +113,115 @@ func (a *AppDataV1) LoadData(rootDir string) error {
+		Environment          string                 `json:"environment,omitempty"`
+		Security             map[string]interface{} `json:"security"`
+		Hosting              map[string]interface{} `json:"hosting,omitempty"`
+		CustomUserDataConfig map[string]interface{} `json:"custom_user_data_config"`
+		Sync                 map[string]interface{} `json:"sync"`
+	}{
+		ConfigVersion:        a.ConfigVersion(),
+		ID:                   a.ID(),
+		Name:                 a.Name(),
+		Location:             a.Location(),
+		DeploymentModel:      a.DeploymentModel(),
+		Environment:          a.Environment,
+		Security:             a.Security,
+		CustomUserDataConfig: a.CustomUserDataConfig,
+		Sync:                 a.Sync,
+	}
+	return MarshalJSON(temp)
+}
+
+// WriteData will write the local Realm app data to disk
+func (a *AppDataV1) WriteData(rootDir string) error {
+	if err := writeSecrets(rootDir, a.Secrets); err != nil {
+		return err
+	}
+	if err := writeEnvironments(rootDir, a.Environments); err != nil {
+		return err
+	}
+	if err := writeValues(rootDir, a.Values); err != nil {
+		return err
+	}
+	if err := writeGraphQL(rootDir, a.GraphQL); err != nil {
+		return err
+	}
+	if err := writeServices(rootDir, a.Services); err != nil {
+		return err
+	}
+	if err := writeFunctionsV1(rootDir, a.Functions); err != nil {
+		return err
+	}
+	if err := writeAuthProviders(rootDir, a.AuthProviders); err != nil {
+		return err
+	}
+	if err := writeTriggers(rootDir, a.Triggers); err != nil {
+		return err
+	}
+	return nil
+}
+
+func writeFunctionsV1(rootDir string, functions []map[string]interface{}) error {
+	if err := os.MkdirAll(filepath.Join(rootDir, NameFunctions), os.ModePerm); err != nil {
+		return err
+	}
+	for _, function := range functions {
+		configFunction := function[NameConfig].(map[string]interface{})
+		nameFunction := configFunction["name"].(string)
+		dataFunction, err := MarshalJSON(configFunction)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameFunctions, nameFunction, FileConfig.String()), 0666, bytes.NewReader(dataFunction))
+		if err != nil {
+			return err
+		}
+		src := function[NameSource].(string)
+		err = WriteFile(filepath.Join(rootDir, NameFunctions, nameFunction, FileSource.String()), 0666, bytes.NewReader([]byte(src)))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeAuthProviders(rootDir string, authProviders []map[string]interface{}) error {
+	if err := os.MkdirAll(filepath.Join(rootDir, NameAuthProviders), os.ModePerm); err != nil {
+		return err
+	}
+	for _, ap := range authProviders {
+		dataProvider, err := MarshalJSON(ap)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameAuthProviders, fmt.Sprintf("%s%s", ap["name"], extJSON)), 0666, bytes.NewReader(dataProvider))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeTriggers(rootDir string, triggers []map[string]interface{}) error {
```

Comment:
should this be in `structure.go` and used in `structure_v2` as well?

Reply:
Triggers is v1 only. It got combined into function in functions in v2.

Reply:
I was incorrect. Fixing in next commit.